bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -40,14 +35,20 @@ impl WitnessProgram {
         if version == WitnessVersion::V0 && (program.len() != 20 && program.len() != 32) {
             return Err(InvalidSegwitV0Length(program.len()));
         }
-        Ok(WitnessProgram { version, program })
+        Ok(WitnessProgram(program))
     }
 
-    /// Returns the witness program version.
-    pub fn version(&self) -> WitnessVersion { self.version }
+    /// Returns the witness program as a [`PushBytes`] reference.
+    pub fn as_push_bytes(&self) -> &PushBytes { &self.0 }
+
+    /// Returns the witness program as a slice reference.
+    pub fn as_bytes(&self) -> &[u8] { self.as_push_bytes().as_bytes() }
+
+    /// The length of the witness program.
+    pub fn len(&self) -> usize { self.as_bytes().len() }
 
-    /// Returns the witness program.
-    pub fn program(&self) -> &PushBytes { &self.program }
+    /// Always false, program is between 2 and 40 bytes inclusive.
+    pub fn is_empty(&self) -> bool { false }
```

Comment:
I would prefer to delete this and instead `allow` the clippy lint.

Reply:
Done.